pkg/lib/cockpit-components-inline-notification.tsx

```diff
@@ -26,17 +26,30 @@ import './cockpit-components-inline-notification.css';
 
 const _ = cockpit.gettext;
 
-function mouseClick(fun) {
-    return function (event) {
+function mouseClick(fun: (ev: React.MouseEvent) => void): (ev: React.MouseEvent) => void {
+    return function (event: React.MouseEvent) {
         if (!event || event.button !== 0)
             return;
         event.preventDefault();
         return fun(event);
     };
 }
 
-export class InlineNotification extends React.Component {
-    constructor(props) {
+interface InlineNotificationProps {
+    text: string;
+    detail?: string;
+    type?: "danger" | "success" | "warning" | "info" | "custom";
```

Comment:
I feel like there is some way to hit this with `typeof AlertProps['variant']` or so.  @jelly made a similar request to me in another PR recently...

Reply:
That works, thanks!